src/sim.rs

```diff
@@ -282,6 +298,7 @@ mod test {
         time::Duration,
     };
 
+    use futures::future;
```

Comment:
Should use the ones from std https://doc.rust-lang.org/std/future/fn.pending.html